### PR TITLE
Add Makefile to Gluster Block Provisioner.

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

**Comment by humblec** [comment]
@wongma7 PTAL.

File: gluster/block/README.md [hunk 1]
```diff
@@ -1,31 +1,37 @@
 # glusterblock Volume Provisioner for Kubernetes 1.5+
```
Comment (obnoxxx) [inline gluster/block/README.md]: Why is there a readme overhaul in a commit called "More divided code for guster-block and heketi mode"? This should be a commit of its own. (Commit messages should always reflect what a commit actually does... ;-).

File: gluster/block/README.md [hunk 2]
```diff
@@ -1,31 +1,37 @@
...
 quay.io/external_storage/glusterblock-provisioner:latest
 ```
 
-# Test instruction
+# Build Gluster Block Provisioner
+
+If you have `go` envrionment configured, you can build your own binary or container image as shown below:
 
-* Build glusterblock-provisioner and container image
+#### Clone and build binary and docker image
```
Comment (obnoxxx) [inline gluster/block/README.md]: Why a level-4 heading after a level-2 heading? In PR #123, I use level 2, and this looks very consistent to me.

File: gluster/block/README.md [hunk 3]
```diff
@@ -1,31 +1,37 @@
...
-[root@localhost]#go build glusterblock-provisioner.go
-[root@localhost]#docker build -t glusterblock-provisioner .
+[root@localhost]# git clone https://github.com/kubernetes-incubator/external-storage
+[root@localhost]# cd external-storage/gluster/block/cmd/glusterblock-provisioner
+[root@localhost]# go build glusterblock-provisioner.go
+[root@localhost]# docker build -t glusterblock-provisioner .
 ```
 
-* Start Kubernetes local cluster
+#### Start Kubernetes local cluster
```
Comment (obnoxxx) [inline gluster/block/README.md]: level 4 ?
also, why is this sorted in under level-1 headline "Build Gluster Block Provisioner"? This does not seem logical.

File: gluster/block/README.md [hunk 4]
```diff
@@ -1,31 +1,37 @@
...
 
-* Start glusterblock provisioner
+#### Start glusterblock provisioner
```
Comment (obnoxxx) [inline gluster/block/README.md]: level 4.
does not belong under "Build Gluster ..."

File: gluster/block/README.md [hunk 5]
```diff
@@ -1,31 +1,37 @@
...
 
 The following example uses `glusterblock-provisioner-1` as the identity for the instance and assumes kubeconfig is at `/root/.kube`. The identity should remain the same if the provisioner restarts. If there are multiple provisioners, each should have a different identity.
 
 ```bash
-docker run -ti -v /root/.kube:/kube -v /var/run/kubernetes:/var/run/kubernetes --privileged --net=host  glusterblock-provisioner /usr/local/bin/glusterblock-provisioner -master=http://127.0.0.1:8080 -kubeconfig=/kube/config -id=glusterblock-provisioner-1
+[root@localhost]#docker run -ti -v /root/.kube:/kube -v /var/run/kubernetes:/var/run/kubernetes --privileged --net=host  glusterblock-provisioner /usr/local/bin/glusterblock-provisioner -master=http://127.0.0.1:8080 -kubeconfig=/kube/config -id=glusterblock-provisioner-1
```
Comment (obnoxxx) [inline gluster/block/README.md]: space after prompt

File: gluster/block/README.md [hunk 6]
```diff
@@ -55,23 +77,39 @@ parameters:
...
+# Gluster-Block Mode parameters:
 
-*`hacount`: This is the count of number of paths to the block target server. This provide high availability via multipathing capability of iscsi. If there is a path failure, the I/Os will not be disturbed and will be served via another available paths.
+If provisioner want to operate on `gluster-block`, below args are required to be filled in storageclass.
 
-* Create a claim
+* `blockmodeargs`:  
+
+This mode requires `glustervol` name and `hosts` to be mentioned in `,` seperated values as shown below. This is a mandatory parameter to be filled
+in storage class parameter.
```
Comment (obnoxxx) [inline gluster/block/README.md]: strange line break

File: gluster/block/README.md [hunk 7]
```diff
@@ -55,23 +77,39 @@ parameters:
...
+
+```
+blockmodeargs: "glustervol=blockmaster1,hosts=10.67.116.108"
+```
+
+# Test Provisioner:
+
+#### Create a claim
```
Comment (obnoxxx) [inline gluster/block/README.md]: level 4 under level 1?

File: gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go [hunk 8]
```diff
@@ -38,23 +38,23 @@ import (
 	"k8s.io/client-go/pkg/api/v1"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
-	//k8s.io/kubernetes/pkg/volume
-	//storage "k8s.io/client-go/pkg/apis/storage/v1beta1"
```
Comment (obnoxxx) [inline gluster/block/cmd/glusterblock-provisioner/glusterblock-provisioner.go]: Ok. The github review feature is not very helpful for per-patch comments... My last comment that requested splitting of a patch to reflect what the commit message acually describes was meant for this commit ...

File: gluster/block/Makefile [hunk 9]
```diff
@@ -0,0 +1,69 @@
+# Copyright 2017 The Kubernetes Authors.
```
Comment (obnoxxx) [inline gluster/block/Makefile]: This is the only of the 4 commits in this PR that actually does what the PR title says: add a makefile.
I guess the other commits should be separate PRs?

**Comment by humblec** [comment]
I think the main concern here is, having more commits in the same PR and readme format. Addressing it in seperate PRs.

**Comment by obnoxxx** [comment]
@humblec, there is nothing wrong with having multiple commits in a single PR. My main concerns are:
* Commits in a PR that have no obvious logical connection (Just one of the commits in a PR matching the PR description). (This is about cleanliness of PRs.)
* Commits that do different things or more things than the commit message tells (examle: readme overhaul as part of a commit that claims to do "more divided code", or a lot of unrelated different code  changes in a commit that claims to fix a calculation). (This is about cleanliness of commits.)

These are more cosmetic, but I think it's important to wrap the actual code changes into good commits and good PRs. This will make it much more easy to do good reviews, to understand the patches, also later when they have been merged.

And yes, on the content, I only had a few comments on the README so far. :-)

**Comment by humblec** [comment]
@obnoxxx at present the test takes ages to complete. That is one of the reason for coupling more commits in one PR and I dont think I can add all the commit headings or message to the Github PR title, that something github picks. The only option is to manually edit all the commit messages in PR title. But, Any way I am seperating the commits and PRs.

**Comment by obnoxxx** [comment]
@humblec sure, usually you want to manually edit the PR message to just describe the overall topic of the PR. No need to list each and every single commit. And if you want to put a lot of independent patches into a single PR because the tests take sooo long (a weak reason actually, but I can somewhat understand the pain), then you just call the PR something like "A couple of improvements" or "Various assorted fixes" ... or similar. Not very good conceptually, but at least clear and not confusing. :-)

**Comment by humblec** [comment]
@wongma7 please review this PR and do the needful.  10 more PRs to come based on this, so the request :)

**Comment by wongma7** [comment]
lgtm. THanks @obnoxxx, agree with the sentiment of making PRs and commits clear & distinct. Tests take a while, but tests are not the critical path anyway, I am :)

**Comment by humblec** [comment]
Thanks @wongma7 💯 .
